paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/FlinkSinkBuilder.java

```diff
@@ -108,9 +108,13 @@ public DataStreamSink<?> build() {
     private DataStreamSink<?> buildDynamicBucketSink(
             DataStream<InternalRow> input, boolean globalIndex) {
         checkArgument(logSinkFunction == null, "Dynamic bucket mode can not work with log system.");
-        return globalIndex
-                ? new GlobalDynamicBucketSink(table, overwritePartition).build(input, parallelism)
-                : new RowDynamicBucketSink(table, overwritePartition).build(input, parallelism);
+        return overwritePartition != null
```

Comment:
Here should only work for `compact & !globalIndex`

Reply:
Added a todo here for `Global Index Sort Compact`, because Global Index table sort compact will need extra tests and   
and end-to-end tests to guarantee it works.

Reply:
done this judge